code/src/pages/cart/components/cart-item/CartItem.tsx

```diff
@@ -21,7 +22,10 @@ const CartItem = ({ item, onRemove, onQuantityChange }: CartItemProps) => {
   const lastDebouncedQuantityRef = useRef(debouncedQuantity);
 
   useEffect(() => {
-    if (debouncedQuantity !== lastDebouncedQuantityRef.current) {
+    if (
+      debouncedQuantity !== lastDebouncedQuantityRef.current &&
+      debouncedQuantity > 0
```

Comment:
move it to a separate variable to increase the readability

Reply:
> move it to a separate variable to increase the readability

I created separate fn because we need to check it each time when `debouncedQuantity` changed
Fixed